src/renderer/dx/DxFontInfo.cpp

```diff
@@ -0,0 +1,382 @@
+    if (!exists)
+    {
+        localeName = L"en-us";
+        THROW_IF_FAILED(familyNames->FindLocaleName(localeName.data(), &index, &exists));
+    }
+
+    // If it still doesn't exist, we're going to try index 0.
+    if (!exists)
+    {
+        index = 0;
+
+        // Get the locale name out so at least the caller knows what locale this name goes with.
+        UINT32 length = 0;
+        THROW_IF_FAILED(familyNames->GetLocaleNameLength(index, &length));
+        localeName.resize(length);
+
+        // https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritelocalizedstrings-getlocalenamelength
+        // https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritelocalizedstrings-getlocalename
+        // GetLocaleNameLength does not include space for null terminator, but GetLocaleName needs it so add one.
+        THROW_IF_FAILED(familyNames->GetLocaleName(index, localeName.data(), length + 1));
+    }
+
+    // OK, now that we've decided which family name and the locale that it's in... let's go get it.
+    UINT32 length = 0;
+    THROW_IF_FAILED(familyNames->GetStringLength(index, &length));
+
+    // Make our output buffer and resize it so it is allocated.
+    std::wstring retVal;
+    retVal.resize(length);
+
+    // FINALLY, go fetch the string name.
+    // https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritelocalizedstrings-getstringlength
+    // https://docs.microsoft.com/en-us/windows/win32/api/dwrite/nf-dwrite-idwritelocalizedstrings-getstring
+    // Once again, GetStringLength is without the null, but GetString needs the null. So add one.
+    THROW_IF_FAILED(familyNames->GetString(index, retVal.data(), length + 1));
+
+    // and return it.
+    return retVal;
+}
+
+// Routine Description:
+// - Creates a DirectWrite font collection of font files that are sitting next to the running
+//   binary (in the same directory as the EXE).
+// Arguments:
+// - dwriteFactory - The DWrite factory to use
+// Return Value:
+// - DirectWrite font collection. May be null if one cannot be created.
+[[nodiscard]] const Microsoft::WRL::ComPtr<IDWriteFontCollection1>& DxFontInfo::NearbyCollection(gsl::not_null<IDWriteFactory1*> dwriteFactory) const
+{
+    // Magic static so we only attempt to grovel the hard disk once no matter how many instances
+    // of the font collection itself we require.
+    static const auto knownPaths = s_GetNearbyFonts();
+
+    // The convenience interfaces for loading fonts from files
+    // are only available on Windows 10+.
+    // Don't try to look up if below that OS version.
+    static const bool s_isWindows10OrGreater = IsWindows10OrGreater();
+
+    if (s_isWindows10OrGreater && !_nearbyCollection)
+    {
+        // Factory3 has a convenience to get us a font set builder.
+        ::Microsoft::WRL::ComPtr<IDWriteFactory3> factory3;
+        THROW_IF_FAILED(dwriteFactory->QueryInterface<IDWriteFactory3>(&factory3));

```

Comment:
This should be the only place where "something" changed after merging #9734. I can't use `As()` here because `dwriteFactory` is a raw pointer.

Reply:
That's fine. Same thing underneath anyway.